timeseries/src/autogluon/timeseries/models/autogluon_tabular/mlforecast.py

```diff
@@ -0,0 +1,301 @@
+        tabular_hyperparameters = self._get_model_params().get(
+            "tabular_hyperparameters", self.default_tabular_hyperparameters
+        )
+        estimator = TabularEstimator(
+            predictor_init_kwargs={
+                "path": self.path,
+                "problem_type": ag.constants.REGRESSION,
+                "eval_metric": self.TIMESERIES_METRIC_TO_TABULAR_METRIC[self.eval_metric],
+                "verbosity": verbosity - 2,
+            },
+            predictor_fit_kwargs={
+                "time_limit": time_limit,
+                "hyperparameters": tabular_hyperparameters,
+                "tuning_data": tuning_data,
+            },
+        )
+        self.mlf.models = {"mean": estimator}
+
+        X_train, y_train = self._get_features_dataframe(train_data, return_X_y=True)
+        with statsmodels_warning_filter():
+            self.mlf.fit_models(X_train, y_train)
+
+        # Use residuals to compute quantiles
+        val_data_future = val_data.slice_by_timestep(-self.prediction_length, None)
+        val_forecast = self._predict_without_quantiles(train_data, val_data_future.drop(self.target, axis=1))
+        residuals = val_forecast["mean"] - val_data_future[self.target]
+        for q in self.quantile_levels:
+            self.quantile_adjustments[q] = np.quantile(residuals, q)
+
+    def _predict_without_quantiles(
+        self,
+        data: TimeSeriesDataFrame,
+        known_covariates: TimeSeriesDataFrame = None,
+    ) -> pd.DataFrame:
+        """Generate a point forecast of the future values using MLForecast.
+
+        Returns
+        -------
+        predictions : pd.DataFrame
+            Predictions with a single column "mean" containing the point forecast.
+        """
+        new_data = self._to_mlforecast_df(data, data.static_features)
+        if known_covariates is not None:
+            dynamic_dfs = [self._to_mlforecast_df(known_covariates, data.static_features, include_target=False)]
+        else:
+            dynamic_dfs = None
+        with statsmodels_warning_filter():
+            raw_predictions = self.mlf.predict(
+                horizon=self.prediction_length,
+                new_data=new_data,
+                dynamic_dfs=dynamic_dfs,
+            )
+        predictions = raw_predictions.rename(columns={"unique_id": ITEMID, "ds": TIMESTAMP})
+        forecast_index = get_forecast_horizon_index_ts_dataframe(data, self.prediction_length)
+        return predictions.set_index([ITEMID, TIMESTAMP]).reindex(forecast_index)
+
+    def predict(
+        self,
+        data: TimeSeriesDataFrame,
+        known_covariates: TimeSeriesDataFrame = None,
+        **kwargs,
+    ) -> TimeSeriesDataFrame:
+        predictions = self._predict_without_quantiles(data, known_covariates)
+        for q in self.quantile_levels:
+            predictions[str(q)] = predictions["mean"] + self.quantile_adjustments[q]
```

Comment:
Wondering how the quantile adjustment is done

Reply:
Here we just use the empirical distribution of the quantiles to generate dummy quantiles. I will add another tree-based model that is trained to predict the quantiles using the wQL loss in a future PR.